server.js

```diff
@@ -1,27 +1,78 @@
-var express = require('express'),
+const express = require('express'),
   // NOTE: require in our request proxy module
   requestProxy = require('express-request-proxy'),
   port = process.env.PORT || 3000,
-  app = express();
+  app = express()
 
+const twilio = require('twilio')
+const bodyParser = require('body-parser')
 
-  var proxyGoogle = function(request, response) {
-    console.log('Routing GoogleMap request for ', request.params[0]);
-    (requestProxy({
-      url: 'https://maps.googleapis.com/' + request.params[0],
-      headers: {Authorization: 'token ' + process.env.MAP_API_KEY}
-    }))(request, response);
-  };
+app.use(bodyParser.urlencoded({ extended: true }))
+app.use(express.static('./'))
+const eventModel = require('./scripts/zipLookupSMS.js')
 
-// app.get('/maps/api/geocode/*', proxyGoogle);
+app.get('*', function (request, response) {
+  console.log('New request:', request.url)
+  response.sendFile('index.html', { root: '.' })
+})
 
-app.use(express.static('./'));
+// steps is the list of callbacks a user will progress through
+const steps = {
+  0: initial,
+  1: zip,
+  2: bye,
+}
 
-app.get('*', function(request, response) {
-  console.log('New request:', request.url);
-  response.sendFile('index.html', { root: '.' });
-});
+// stores steps a user is at
+const storage = {}
 
-app.listen(port, function() {
-  console.log('Server started on port ' + port + '!');
-});
+// TODO: Decide on actual flow, maybe text the closest five?
+app.post('/sms', (req, res) => {
+  const from = req.body.From
+  const session = storage[from] || { step: 0 }
+  steps[session.step](req, res, session).then(function (twiml) {
+    session.step ++
+    storage[from] = session
+    console.log('session:', session)
+    console.log('storage:', storage)
+    res.writeHead(200, { 'Content-Type': 'text/xml' })
+    res.end(twiml.toString())
+  })
+})
+
+function initial (req, res, session) {
+  const twiml = new twilio.TwimlResponse()
+  twiml.message('hey there, whats your zip?')
+  session.campaign = req.body.Body
+  return new Promise(function (resolve, reject) {
+    resolve(twiml)
```

Comment:
Maybe also when figuring out the flow and how many we text we want to figure out catch blocks for these promises.